### PR TITLE
IDT - BossMobile

### PR DESCRIPTION
**Pull request by pablopzsj**
Implement a scenario: add credit card

### CODE REVIEW

File: IDT.BossMobile/src/test/java/com/jalasoft/steps/settings/PaymentMethodSteps.java [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+
+import cucumber.api.java.en.And;
+import cucumber.api.java.en.Then;
+import cucumber.api.java.en.When;
+
+import java.util.Map;
+
+import static org.testng.Assert.assertTrue;
+
+public class PaymentMethodSteps {
```
Comment (angy113) [inline IDT.BossMobile/src/test/java/com/jalasoft/steps/settings/PaymentMethodSteps.java]: Comment class

File: IDT.BossMobile/src/test/resources/settings/addCreditCard.feature [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+      And I enter the credit card information with the following data:
+        | CardNumber | 5454545454545454 |
+        | Year       | 2019             |
+        | Month      | 11 - November    |
+        | NameCard   | Pablo Ramirez    |
+        | Street     | Simon Lopez      |
+        | City       | Cochabamba       |
+        | State      | CO               |
+        | Zip        | 00000            |
+    And I press Save button of Payments Methods
```
Comment (angy113) [inline IDT.BossMobile/src/test/resources/settings/addCreditCard.feature]: Add tab to and

File: IDT.BossMobile/src/main/java/com/jalasoft/bossmobile/pages/settings/PaymentsMethodsPage.java [hunk 3]
```diff
@@ -0,0 +1,214 @@
...
+        final Map<CreditCardEnum, Steps> strategyMap = new HashMap<>();
+        strategyMap.put(CARDNUMBER, () -> setCreditCardNumber(valuesCreditCard.get(CARDNUMBER)));
+        strategyMap.put(YEAR, () -> selectYear(valuesCreditCard.get(YEAR)));
+        strategyMap.put(MONTH, () -> selectMonth(valuesCreditCard.get(MONTH)));
+        strategyMap.put(NAMECARD, () -> setNameCard(valuesCreditCard.get(NAMECARD)));
+        strategyMap.put(STREET, () -> setStreet(valuesCreditCard.get(STREET)));
+        strategyMap.put(CITY, () -> setCity(valuesCreditCard.get(CITY)));
+        strategyMap.put(STATE, () -> setState(valuesCreditCard.get(STATE)));
+        strategyMap.put(ZIP, () -> setZIP(valuesCreditCard.get(ZIP)));
+        nameCreditCard = valuesCreditCard.get(NAMECARD);
```
Comment (angy113) [inline IDT.BossMobile/src/main/java/com/jalasoft/bossmobile/pages/settings/PaymentsMethodsPage.java]: Why is this assignation for?

File: IDT.BossMobile/src/test/java/com/jalasoft/steps/settings/PaymentMethodSteps.java [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+     */
+    public PaymentMethodSteps(final PaymentsMethodsPage paymentsMethodsPage, final SettingsPage settingsPage) {
+        this.paymentsMethodsPage = paymentsMethodsPage;
+        this.settingsPage = settingsPage;
+    }
+
+    /**
+     * Go to payments methods page.
+     */
+    @And("^I press Payments Methods button$")
```
Comment (angy113) [inline IDT.BossMobile/src/test/java/com/jalasoft/steps/settings/PaymentMethodSteps.java]: Replace @And for @When

File: IDT.BossMobile/src/test/java/com/jalasoft/steps/settings/PaymentMethodSteps.java [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+    public void iPressAddCardButton() {
+        paymentsMethodsPage.clickOnAddCard();
+    }
+
+    /**
+     * Set the credit card information.
+     *
+     * @param valuesCreditCard credit card values.
+     */
+    @And("^I enter the credit card information with the following data:$")
```
Comment (angy113) [inline IDT.BossMobile/src/test/java/com/jalasoft/steps/settings/PaymentMethodSteps.java]: Replace @And for @When

File: IDT.BossMobile/src/main/java/com/jalasoft/bossmobile/pages/settings/PaymentsMethodsPage.java (hunk 3 above)
Comment (pablopzsj) [reply IDT.BossMobile/src/main/java/com/jalasoft/bossmobile/pages/settings/PaymentsMethodsPage.java]: I need save the name credit card

**Pull request merged**